File: frontend/src/lib/languages.ts
// Languages the Nemotron streaming model can be prompted with. Mirrors the
// server-side list (see server/app/languages.py); the live list is fetched
// from /health at runtime, this is the fallback used before that resolves.

export interface LanguageInfo {
  code: string
  name: string
}

// "auto" lets the model detect and tag the spoken language.
export const DEFAULT_LANGUAGE = 'auto'

export const LANGUAGES: LanguageInfo[] = [
  { code: 'auto', name: 'Auto-detect' },
  { code: 'en-US', name: 'English' },
  { code: 'es-ES', name: 'Spanish' },
  { code: 'fr-FR', name: 'French' },
  { code: 'de-DE', name: 'German' },
  { code: 'it-IT', name: 'Italian' },
  { code: 'pt-PT', name: 'Portuguese' },
  { code: 'nl-NL', name: 'Dutch' },
  { code: 'pl-PL', name: 'Polish' },
  { code: 'sv-SE', name: 'Swedish' },
  { code: 'cs-CZ', name: 'Czech' },
  { code: 'sk-SK', name: 'Slovak' },
  { code: 'da-DK', name: 'Danish' },
  { code: 'nb-NO', name: 'Norwegian' },
  { code: 'fi-FI', name: 'Finnish' },
  { code: 'hu-HU', name: 'Hungarian' },
  { code: 'ro-RO', name: 'Romanian' },
  { code: 'hr-HR', name: 'Croatian' },
  { code: 'bg-BG', name: 'Bulgarian' },
  { code: 'et-EE', name: 'Estonian' },
  { code: 'uk-UA', name: 'Ukrainian' },
  { code: 'ru-RU', name: 'Russian' },
  { code: 'tr-TR', name: 'Turkish' },
  { code: 'ar', name: 'Arabic' },
  { code: 'hi-IN', name: 'Hindi' },
  { code: 'ja-JP', name: 'Japanese' },
  { code: 'ko-KR', name: 'Korean' },
  { code: 'vi-VN', name: 'Vietnamese' },
  { code: 'zh-CN', name: 'Mandarin Chinese' },
]

File: server/app/languages.py
"""Languages understood by the Nemotron streaming ASR model.

The model is prompt-conditioned on a target language given as a BCP-47 locale
(e.g. ``en-US``), or the special value ``auto`` to auto-detect the spoken
language.  The list below is a curated subset of the model's supported locales;
``auto`` is always available and is the default.
"""
from __future__ import annotations

# Special value: let the model detect the language itself.
AUTO = "auto"

# Default used when the caller omits a language.
DEFAULT_LANGUAGE = AUTO

# locale code -> human-readable name. Ordered for dropdown display.
LANGUAGE_NAMES: dict[str, str] = {
    "en-US": "English",
    "es-ES": "Spanish",
    "fr-FR": "French",
    "de-DE": "German",
    "it-IT": "Italian",
    "pt-PT": "Portuguese",
    "nl-NL": "Dutch",
    "pl-PL": "Polish",
    "sv-SE": "Swedish",
    "cs-CZ": "Czech",
    "sk-SK": "Slovak",
    "da-DK": "Danish",
    "nb-NO": "Norwegian",
    "fi-FI": "Finnish",
    "hu-HU": "Hungarian",
    "ro-RO": "Romanian",
    "hr-HR": "Croatian",
    "bg-BG": "Bulgarian",
    "et-EE": "Estonian",
    "uk-UA": "Ukrainian",
    "ru-RU": "Russian",
    "tr-TR": "Turkish",
    "ar": "Arabic",
    "hi-IN": "Hindi",
    "ja-JP": "Japanese",
    "ko-KR": "Korean",
    "vi-VN": "Vietnamese",
    "zh-CN": "Mandarin Chinese",
}

# Full ordered list exposed to clients: Auto-detect first, then the locales.
LANGUAGES: list[tuple[str, str]] = [(AUTO, "Auto-detect")] + list(
    LANGUAGE_NAMES.items()
)

_VALID = {AUTO, *LANGUAGE_NAMES}


def is_supported(code: str | None) -> bool:
    """Whether ``code`` is a language this model can be prompted with."""
    return code is None or code in _VALID


def resolve_language(code: str | None, default: str) -> str:
    """Return a valid language prompt value, falling back to ``default``."""
    if code and code in _VALID:
        return code
    return default
